docs/09_firebase_deployment.md

```diff
@@ -0,0 +1,201 @@
+It's time to deploy your Flutter application to the Firebase Hosting!
+ 
+## Building and deploying the application to the Firebase Hosting
+
+### Preparing your environment 
+Before deploying metrics application, make sure you have the correct Flutter version installed,
+ by running the `flutter --version` command. You should have `v1.15.3` installed. 
+
+To switch the Flutter to the required version, you should run the `flutter version 1.15.3` command.
+
+Also, you should enable flutter web support by running the `flutter config --enable-web` command.
+
+### Building and deploying Flutter application
+ 
+1. Open the terminal and navigate to the metrics project folder.
+2. Run `firebase login` command and follow the instructions to log in to the Firebase CLI with your Google account.
+ Use the same account that you used to create your Firebase project (or the one that has access to it).
+3. After you have logged in, run the `firebase use --add` command, and select the project id of the project, created in previous steps.
+4. Give an alias to your project.
+5. Run the `flutter build web` command from the root of the metrics project to build the release version of the application.
+ It is recommended to add `--dart-define=FLUTTER_WEB_USE_SKIA=true` parameter to build the application with the `SKIA` renderer.
+4. Run the `firebase deploy --only hosting` command to deploy an application to the Firebase Hosting.
+
+After the deployment process finished, your application will be accessible using the `Hosting URL`, printed to console.
+
+## Creating test data for the deployed application
+
+Once you've deployed the metrics application to Firebase Hosting, you can create a sample data to test the application.  
+The application reads data from 2 Firestore collections: `projects` and `build`. To create a sample data follow the next steps: 
+
+1. Deploying the `seedData` cloud function: 
+    1. Go to the `metrics/firebase/index.js` file and comment this line of code: `return resp.status(200).send('done');`.
+    2. Open console, navigate to `metrics` directory and run the `firebase deploy --only functions`
+     command to deploy this cloud function.
+    3. Once the function deployment is finished, the URL will be printed to the console.
+     You can trigger this function to create builds from the project, using this URL.  
+     This HTTP function has the following query parameters: 
+        - buildsCount (required) - the number of builds to generate.
+        - projectId (required) - the project identifier for which builds will be generated.
+        - startDate (optional) - builds will be generated with 'startedAt'
+         property in the range from the `startDate` to `startDate - 7` days. Defaults to the current date.
+        - delay (optional) - is the delay in milliseconds between adding builds to the project.
+
+2. Creating projects in Firestore Database:
```

Comment:
Maybe it would be better to swap 1 & 2? First, create a project in the Console - then create seed data?